### PR TITLE
Add a command to delete old gift cards

### LINKED ISSUE #228: Create a command to delete old gift cards
It would be handy to have a command that would delete expired gift cards automatically (where `expiresAt` is in the the past)

Also, since we would need that feature in 0.11, it would make sense to base this on `createdAt` property for that branch

### PR DESCRIPTION
**Pull request by Roshyo**
Fixes #228

### CODE REVIEW

File: src/Doctrine/ORM/GiftCardRepository.php [hunk 1]
```diff
@@ -69,4 +69,14 @@ public function createAccountListQueryBuilder(CustomerInterface $customer): Quer
 
         return $qb;
     }
+
+    public function findCreatedBefore(\DateTimeInterface $date): array
+    {
+        $qb = $this->createQueryBuilder('gc');
+
+        $qb->andWhere('gc.createdAt < :date');
+        $qb->setParameter('date', $date->format('Y-m-d H:i:s'));
```
Comment (loevgaard) [inline src/Doctrine/ORM/GiftCardRepository.php]:
```suggestion
        $qb->setParameter('date', $date);
```

Isn't this possible?

File: src/Command/Cli/DeleteOldGiftCardsCommand.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    protected static $defaultName = 'setono:sylius-gift-card:delete-old-gift-cards';
+
+    private GiftCardRepositoryInterface $giftCardRepository;
+
+    private ObjectManager $giftCardManager;
+
+    public function __construct(
+        GiftCardRepositoryInterface $giftCardRepository,
+        ObjectManager $giftCardManager,
+        ?string $name = null
```
Comment (loevgaard) [inline src/Command/Cli/DeleteOldGiftCardsCommand.php]:
```suggestion
```

Why do you need the name?

Comment (Roshyo) [reply src/Command/Cli/DeleteOldGiftCardsCommand.php]: To call parent constructor

File: src/Command/Cli/DeleteOldGiftCardsCommand.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        } catch (\Exception $e) {
+            $output->writeln('The provided date or period is not valid.');
+
+            return 0;
+        }
+
+        $giftCards = $this->giftCardRepository->findCreatedBefore($thresholdDate);
+        foreach ($giftCards as $giftCard) {
+            $this->giftCardManager->remove($giftCard);
```
Comment (loevgaard) [inline src/Command/Cli/DeleteOldGiftCardsCommand.php]: I think this command should instead disable the gift cards and not remove them

File: src/Doctrine/ORM/GiftCardRepository.php [hunk 4]
```diff
@@ -69,4 +69,14 @@ public function createAccountListQueryBuilder(CustomerInterface $customer): Quer
 
         return $qb;
     }
+
+    public function findCreatedBefore(\DateTimeInterface $date): array
```
Comment (loevgaard) [inline src/Doctrine/ORM/GiftCardRepository.php]:
```suggestion
    public function findCreatedBefore(\DateTimeInterface $date, int $limit = 100): array
```
Else we could run into memory issues

Comment (Roshyo) [reply src/Doctrine/ORM/GiftCardRepository.php]: I'd advise to allow null so we can get them all if we want to

File: src/Command/Cli/DeleteOldGiftCardsCommand.php (hunk 2 above)
Comment (loevgaard) [reply src/Command/Cli/DeleteOldGiftCardsCommand.php]: Yeah, but it's null by default, so you don't need it in your constructor :)

**Review by loevgaard — APPROVED** [review]

**Pull request merged**